src/main/java/bio/terra/service/dataset/DatasetTableDao.java

```diff
@@ -100,18 +110,33 @@ private void createColumns(UUID tableId, Collection<Column> columns) {
                 .rawTableName(rs.getString("raw_table_name"))
                 .softDeleteTableName(rs.getString("soft_delete_table_name"));
 
-            List<String> primaryKey = DaoUtils.getStringList(rs, "primary_key");
             List<Column> columns = retrieveColumns(table);
+            table.columns(columns);
+
             Map<String, Column> columnMap = columns
                 .stream()
                 .collect(Collectors.toMap(Column::getName, Function.identity()));
 
+            List<String> primaryKey = DaoUtils.getStringList(rs, "primary_key");
             List<Column> naturalKeyColumns = primaryKey.stream()
                 .map(columnMap::get)
                 .collect(Collectors.toList());
             table.primaryKey(naturalKeyColumns);
 
-            return table.columns(columns);
+            long bqPartitionVersion = rs.getLong("bigquery_partition_config_version");
+            String bqPartitionConfig = rs.getString("bigquery_partition_config");
+            if (bqPartitionVersion == 1) {
```

Comment:
This check caught that the object mapper wasn't actually persisting all the `private` members (eek!). Added a case to the `DatasetDaoTest` to guard against regressions there.